### PR TITLE
Dependency updates

### PR DESCRIPTION
**Pull request by JamesGlover**
This was fairly straight forward, despite the ruby update.

Most the cop changes are just to remove actions that are redundant in ruby 3 (Sort on the returned files) or actively maintain the old behaviour (Freezing interpolated string literals)

There's also the OpenStruct changes that I've brought in from asset audits.

### CODE REVIEW

**Comment by JamesGlover** [comment]
Oh, and after all that we fail on creating a database. Sigh.

File: db/migrate/20180731142628_add_column_comment_to_qc_result.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 class AddColumnCommentToQcResult < ActiveRecord::Migration
   def change
-    set_column_comment :qc_result, :last_updated, 'The date the qc_result was last updated in SS'
+    change_column_comment :qc_result, :last_updated, from: nil, to: 'The date the qc_result was last updated in SS'
```
Comment (JamesGlover) [inline db/migrate/20180731142628_add_column_comment_to_qc_result.rb:3]: In practice we're unlikely to ever run this again, but this updates it from using the migration_comments syntax to the native rails syntax. migration_comments is abandoned and was blocking the Ruby 3 update.

File: lib/deployed_version.rb [hunk 2]
```diff
@@ -107,15 +107,10 @@ def read_file(filename)
...
   REPO_URL       = REPO_DATA.release_url.presence || '#'
   HOSTNAME       = Socket.gethostname
 
-  require 'ostruct'
-  DETAILS = OpenStruct.new(
-    name: nil,
-    version: VERSION_ID,
-    environment: ENVIRONMENT
-  )
+  DETAILS = { version: VERSION_ID, environment: ENVIRONMENT }.freeze
```
Comment (JamesGlover) [inline lib/deployed_version.rb:115]: Same pattern I've applied elsewhere when running into the same issue.

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -12,7 +12,7 @@
 
 # Requires supporting ruby files with custom matchers and macros, etc,
 # in spec/support/ and its subdirectories.
-Dir[Rails.root.join('spec/support/**/*.rb')].sort.each { |f| require f }
+Dir[Rails.root.join('spec/support/**/*.rb')].each { |f| require f }
```
Comment (JamesGlover) [inline spec/spec_helper.rb:15]: Rubocop: Ruby 3 performs this sort by default. Nicely slaying a whole suite of bugs that manifest differently on the CI vs development.

**Review by KatyTaylor — APPROVED** [review]

**Pull request merged**